dex/pair/src/locked_asset.rs

```diff
@@ -0,0 +1,145 @@
+elrond_wasm::imports!();
+elrond_wasm::derive_imports!();
+
+use crate::config;
+use crate::errors::*;
+
+#[derive(TopEncode, TopDecode, NestedEncode, NestedDecode, TypeAbi, Debug)]
+pub struct LockedAssetAttributes<M: ManagedTypeApi> {
+    pub unlock_epoch: u64,
+    pub amount: BigUint<M>,
+}
+
+#[elrond_wasm::module]
+pub trait LockedAsset: config::ConfigModule + token_send::TokenSendModule {
+    #[endpoint(setLockedAssetTokenIdFirst)]
```

Comment:
how are these tokens issued and roles set for them?

Reply:
the owner of the router SC i think